packages/data-explorer-ui/src/components/Filter/components/SearchAllFilters/searchAllFilters.tsx

```diff
@@ -0,0 +1,173 @@
+import {
+  AutocompleteRenderGroupParams,
+  AutocompleteRenderInputParams,
+  Checkbox,
+  ListItemButton,
+  ListItemText,
+  PaperProps,
+  Popper,
+  PopperProps,
+  Typography,
+} from "@mui/material";
+import React, { Fragment, useMemo } from "react";
+import {
+  SelectCategoryValueView,
+  SelectCategoryView,
+} from "../../../../common/entities";
+import { OnFilterFn } from "../../../../hooks/useCategoryFilter";
+import { CheckedIcon } from "../../../common/CustomIcon/components/CheckedIcon/checkedIcon";
+import { UncheckedIcon } from "../../../common/CustomIcon/components/UncheckedIcon/uncheckedIcon";
+import { FilterNoResultsFound } from "../FilterNoResultsFound/filterNoResultsFound";
+import { SearchAllFiltersMenu } from "../SearchAllFiltersMenu/searchAllFiltersMenu";
+import { SearchAllFiltersSearch } from "../SearchAllFiltersSearch/searchAllFiltersSearch";
+import {
+  GroupDivider,
+  GroupHeading,
+  SearchAllFilters as Autocomplete,
+  SearchAllFiltersMenuPaper,
+} from "./searchAllFilters.styles";
+
+// Special option values, for insterting menu content other than actual options
+enum SPECIAL_OPTION {
+  NO_RESULTS = "SPECIAL_OPTION_NO_RESULTS",
+}
+
+interface SearchAllFiltersProps {
+  categoryViews: SelectCategoryView[];
+  onFilter: OnFilterFn;
+}
+
+interface CategoryValueOption {
+  categoryKey: SelectCategoryView["key"];
+  categoryLabel: SelectCategoryView["label"];
+  value: SelectCategoryValueView;
+}
+
+export type FilterOption = CategoryValueOption | SPECIAL_OPTION;
+
+const PaperComponent = (props: PaperProps): JSX.Element => (
+  <SearchAllFiltersMenuPaper variant="menu" {...props} />
+);
+
+const PopperComponent = ({ modifiers, ...props }: PopperProps): JSX.Element => (
+  <Popper
+    modifiers={[
+      {
+        name: "offset",
+        options: {
+          offset: [0, 8],
+        },
+      },
+      ...(modifiers || []),
+    ]}
+    {...props}
+  />
+);
+
+const renderGroup = ({
+  children,
+  group,
+  key,
+}: AutocompleteRenderGroupParams): JSX.Element =>
+  group ? (
+    <Fragment key={key}>
+      <GroupDivider />
```

Comment:
Try: 
```
{Number(key) !== 0 && <GroupDivider />}
```
Not sure why `AutocompleteRenderGroupParams` returns key as a string? Convert to number and test that the first key is not equal to 0...